lib/exercises.js

```diff
@@ -1,17 +1,61 @@
 // This method will return an array of arrays.
 // Each subarray will have strings which are anagrams of each other
-// Time Complexity: ?
-// Space Complexity: ?
+// Time Complexity: O(n)^2
+// Space Complexity: O(n)^2
 function grouped_anagrams(strings) {
-  throw new Error("Method hasn't been implemented yet!");
+  const anagrams = {};
+  strings.forEach((string) => {
+    const sorted = alphabetize(string);
+    if (anagrams[sorted]) {
+      return anagrams[sorted].push(string);
+    }
+    anagrams[sorted] = [string];
+  })
+  sortedAnagrams = []
+  for (const key in anagrams) {
+    sortedAnagrams.push(anagrams[key])
+  }
+ 
+  return sortedAnagrams
+
 }
 
+function alphabetize(word) {
+  word = word.split('');
+  word = word.sort();
+  word = word.join('');
+  return word;
+}
+
+
 // This method will return the k most common elements
 // in the case of a tie it will select the first occuring element.
-// Time Complexity: ?
-// Space Complexity: ?
+// Time Complexity: O(n)
+// Space Complexity: O(n) where n is number of unique numbers in list
 function top_k_frequent_elements(list, k) {
```

Comment:
For time complexity this is O(n) where n is the number of elements in the list, if there are few distinct numbers.

Or O(n + m log m) where m is the number of distinct numbers (since you're sorting them.